lib/rjgit.rb

```diff
@@ -190,6 +190,40 @@ def self.describe(repository, ref, options = {})
       end
       command.call
     end
+
+    # options:
+    #  * ref
+    #  * path_filter
+    #  * case_insensitive
+    def self.grep(repository, query, options={})
+      case_insensitive = options[:case_insensitive]
+      repo = RJGit.repository_type(repository)
+      walk = RevWalk.new(repo)
+      ls_tree_options = {:recursive => true, :path_filter => options[:path_filter]}
+
+      raise "A #{query.class} was passed to #{self}.grep().  Only Regexps and Strings are supported!" unless String === query || Regexp === query
+
+      if case_insensitive
+        query = case query
+        when Regexp then Regexp.new(query.source, query.options | Regexp::IGNORECASE)
+        when String then query.downcase
+        end
+      end
+
+      ls_tree(repo, nil, options.fetch(:ref, 'HEAD'), ls_tree_options).each_with_object({}) do |item, result|
+        blob = Blob.new(repo, item[:mode], item[:path], walk.lookup_blob(ObjectId.from_string(item[:id])))
+        next if blob.binary?
+
+        rows = blob.data.split("\n")
```

Comment:
What about adding an `&block` optional parameter to this method, and `yield blob.data` or `yield rows` here if `block_given?`?

Reply:
Can you show me an example of a block you might pass to `Gollum::Git.grep`?  I don't see any uses of grep in rjgit_adapter itself and I'm not sure what you have in mind.  It seems that that would require the user to parse the blob data.  Personally I'd rather just pass a Regex to `grep` and let the library figure out how to parse it.